code/modules/mob/mob_helpers.dm

```diff
@@ -676,16 +676,32 @@ proc/is_blind(A)
 		if (istype(location, /turf/simulated))
 			location.add_vomit_floor(src, 1)
 
-		nutrition -= 40
+		nutrition -= 60
 		if (intoxication)//The pain and system shock of vomiting, sobers you up a little
 			intoxication *= 0.8
 
 		if (istype(src, /mob/living/carbon/human))
 			ingested.trans_to_turf(location,30)//Vomiting empties the stomach, transferring 30u reagents to the floor where you vomited
-	else
+	else if (prob(50))
 		src.visible_message("<span class='warning'>[src] retches, attempting to vomit!</span>","<span class='warning'>You gag and collapse as you feel the urge to vomit, but there's nothing in your stomach!</span>")
 		Weaken(4)
 
+/mob/living/carbon/human/proc/delayed_vomit()
+
+	if(!check_has_mouth())
+		return
+	if(stat == DEAD)
+		return
+	if(!lastpuke)
+		lastpuke = 1
+		src << "<span class='warning'>You feel nauseous...</span>"
+		spawn(150)	//15 seconds until second warning
+			src << "<span class='warning'>You feel like you are about to throw up!</span>"
+			spawn(100)	//and you have 10 more for mad dash to the bucket
+				vomit()//Vomit function is in mob helpers
+				spawn(350)	//wait 35 seconds before next volley
+					lastpuke = 0

```

Comment:
Using costly spawn() instead of world time comparisons. Shitty code to start with, but eh.